Source/ASDisplayNode+Layout.mm

```diff
@@ -300,14 +300,16 @@ - (void)_u_measureNodeWithBoundsIfNecessary:(CGRect)bounds
   
   CGSize boundsSizeForLayout = ASCeilSizeValues(bounds.size);
 
-  // Prefer _pendingDisplayNodeLayout over _calculatedDisplayNodeLayout (if exists, it's the newest)
-  // If there is no _pending, check if _calculated is valid to reuse (avoiding recalculation below).
-  if (_pendingDisplayNodeLayout == nullptr || _pendingDisplayNodeLayout->version < _layoutVersion) {
-    if (_calculatedDisplayNodeLayout->version >= _layoutVersion
-        && (_calculatedDisplayNodeLayout->requestedLayoutFromAbove == YES
-            || CGSizeEqualToSize(_calculatedDisplayNodeLayout->layout.size, boundsSizeForLayout))) {
-      return;
-    }
+  // Prefer a newer and not yet applied _pendingDisplayNodeLayout over _calculatedDisplayNodeLayout
+  // If there is no such _pending, check if _calculated is valid to reuse (avoiding recalculation below).
+  BOOL pendingLayoutIsPreferred = (_pendingDisplayNodeLayout != nullptr
+                                   && _pendingDisplayNodeLayout->version >= _layoutVersion
+                                   && _pendingDisplayNodeLayout->version > _calculatedDisplayNodeLayout->version); // _pending is not yet applied
+  BOOL calculatedLayoutIsReusable = (_calculatedDisplayNodeLayout->version >= _layoutVersion
+                                     && (_calculatedDisplayNodeLayout->requestedLayoutFromAbove == YES
+                                         || CGSizeEqualToSize(_calculatedDisplayNodeLayout->layout.size, boundsSizeForLayout)));
+  if (!pendingLayoutIsPreferred && calculatedLayoutIsReusable) {
```

Comment:
Nit: let's take the opportunity to remove `== YES` and maybe add a comment explaining the `requestedLayoutFromAbove + sizeCheck` logic.